MarkBind/markbind
# Upgrade MarkBind dependencies

**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [ ] Documentation update
• [ ] Bug fix
• [ ] New feature
• [ ] Enhancement to an existing feature
• [x] Other, please explain: Dependency update



Part of #461 
Follows #788 
Follows #948



**What is the rationale for this request?**

Our dependencies are slowly getting outdated over time. Some of them are in need of upgrading due to security vulnerabilities reported by `npm audit`. Some already have major version releases — improvements and new bug(fix)s of course — that still appear to be compatible with MarkBind.

**What changes did you make? (Give an overview)**

Used npm 6.13.2 (current latest) to:
1. upgrade vulnerable dependencies reported by `npm audit`.
   - first commit contains changes in package-lock.json only.
   - changes in package.json included in next two commits.
     - eslint-plugin-lodash: 5.1.0 → `^5.1.1` → `^6.0.0`
     - lodash: 4.17.11 → `^4.17.12` (`^4.17.15`)
2. update 24/38 dependencies — 8 major versions, 16 minor/patch versions
3. update 9/9 dev dependencies — 7 major versions, 2 minor versions
   - `eslint` updates required configuring [`arrow-parens` (further reading)](https://eslint.org/docs/rules/arrow-parens#further-reading), and updating 2 files (src/Site.js, src/lib/markbind/src/parser.js) to adhere to [`prefer-object-spread`](https://eslint.org/docs/rules/prefer-object-spread).

**Provide some example code that this change will affect:**

For devs:


```diff
- .map(page => Object.assign({ headings: page.headings }, page.frontMatter)),
+ .map(page => ({ headings: page.headings, ...page.frontMatter })),
```
```diff
- const combinedBases = Object.assign({}, childrenBase, foundBase);
+ const combinedBases = { ...childrenBase, ...foundBase };
```

For users: none.

**Is there anything you'd like reviewers to focus on?**

Do the changes required due to `eslint` updates make sense?

Consider releasing this PR as its own minor version so that it may be easier to check if unintended changes for projects that use MarkBind occur due to other PRs or dependencies updated in this PR.

**Testing instructions:**

- Check that `npm i` using npm 6.12.1 (packaged with Node.js LTS version 12.13.1) or higher doesn't modify package-lock.json.
- Check the output diff on an existing project that uses MarkBind; steps detailed under **Testing instructions** in #948.
